service/src/main/kotlin/io/provenance/explorer/domain/entities/Transactions.kt

```diff
@@ -671,3 +675,84 @@ class TxSingleMessageCacheRecord(id: EntityID<Int>) : IntEntity(id) {
     var txMessageType by TxSingleMessageCacheTable.txMessageType
     var processed by TxSingleMessageCacheTable.processed
 }
+
+object TxGasCacheTable : IntIdTable(name = "tx_gas_cache") {
+    val hash = varchar("hash", 64)
+    val txTimestamp = datetime("tx_timestamp")
+    val gasWanted = integer("gas_wanted")
+    val gasUsed = integer("gas_used")
+    val feeAmount = double("fee_amount")
+    val processed = bool("processed")
+}
+
+class TxGasCacheRecord(id: EntityID<Int>) : IntEntity(id) {
+    companion object : IntEntityClass<TxGasCacheRecord>(TxGasCacheTable) {
+
+        fun insertIgnore(tx: ServiceOuterClass.GetTxResponse, txTime: DateTime) =
+            transaction {
+                if (findByHash(tx.txResponse.txhash) == null) {
+                    // calculate fee amount
+                    val authInfo = tx.txResponse.tx.unpack(TxOuterClass.Tx::class.java).authInfo
+                    val numerator = authInfo.fee.amountList.firstOrNull()?.amount?.toBigInteger() ?: BigInteger.ZERO
+                    val denominator = authInfo.fee.gasLimit.toBigInteger()
+                    val fee = numerator.toDouble().div(denominator.toDouble())
+
+                    TxGasCacheTable.insertIgnore {
+                        it[hash] = tx.txResponse.txhash
+                        it[txTimestamp] = txTime
+                        it[gasUsed] = tx.txResponse.gasUsed.toInt()
+                        it[gasWanted] = tx.txResponse.gasWanted.toInt()
+                        it[feeAmount] = fee
```

Comment:
you want the numerator here. This should be the fee amount paid, not the min gas fee factor.